packages/babel-template/src/index.js

```diff
@@ -77,37 +68,34 @@ let templateVisitor = {
   // 360
   noScope: true,
 
-  enter(path, args) {
-    let { node } = path;
-    if (node[TEMPLATE_SKIP]) return path.skip();
-
-    if (t.isExpressionStatement(node)) {
-      node = node.expression;
-    }
+  Identifier(path, args) {
+    const { name } = path.node;
 
     let replacement;
 
-    if (t.isIdentifier(node) && node[FROM_TEMPLATE]) {
-      if (has(args[0], node.name)) {
-        replacement = args[0][node.name];
-      } else if (node.name[0] === "$") {
-        let i = +node.name.slice(1);
-        if (args[i]) replacement = args[i];
-      }
+    if (has(args[0], name)) {
+      replacement = args[0][name];
+    } else if (name[0] === "$") {
+      const i = +name.slice(1);
+      if (args.length > i) replacement = args[i];
     }
 
-    if (replacement === null) {
-      path.remove();
+    const { parentPath } = path;
+    if (parentPath.isExpressionStatement()) {
+      path = parentPath;
     }
 
-    if (replacement) {
-      replacement[TEMPLATE_SKIP] = true;
+    if (replacement === null) {
+      path.remove();
+    } else if (replacement) {
       path.replaceInline(replacement);
+      path.skip();
```

Comment:
I think this is going to fail if `replacement` is an array of nodes. Not sure what the right fix would be.